EmbeddedPkg/Application/AndroidFastboot/AndroidFastbootApp.c

```diff
@@ -477,6 +483,16 @@ AcceptData (
     SEND_LITERAL ("OKAY");
     mState = ExpectCmdState;
   }
+
+  if (mTransport->RequestReceive) {
+    if (mState == ExpectCmdState) {
+      mTransport->RequestReceive (RECEIVE_CMD_LEN);
+    } else if (mState == ExpectDataState) {
+      mTransport->RequestReceive (RemainingBytes - Size);
+    } else {
+      ASSERT (FALSE);
+    }
+  }

```

Comment:
Why do we need to move the code? If not, what happen?

Reply:
Because of the patchs in OpenPlatformPkg/Drivers/Usb/DwUsb3Dxe，we need to call "mTransport->RequestReceive" every time we want to start a receiving which was doing in OpenPlatformPkg/Drivers/Usb/DwUsb3Dxe before.

Reply:
AndroidFastbootApp is a common application. HiKey, HiKey960 and ArmVExpress are also using AndroidFastbootApp. Do they need to update code when your patch is merged?

Reply:
HiKey and ArmVExpress need to update code if you apply the patch to their branch.
Hikey960 do not need to update code because it share code of OpenPlatformPkg/Drivers/Usb/DwUsb3Dxe with Hikey970.

Reply:
Sorry for replying late, I need some time to have a look at the code of Hikey and ArmVExpress. And I want to know how to verify the code of Hikey and ArmVExpress?

Reply:
You can follow it to build the firmware on HiKey (https://github.com/ARM-software/arm-trusted-firmware/blob/master/docs/plat/hikey.rst). There're some HiKey boards in Hisilicon lab. Since the document is for latest code, you could get a similar one from the same directory in your HiKey970 code base.
I'm not sure whether you have ArmVExpress. Maybe I can help to verify it on ArmVExpress.

Reply:
@chenyu56 Any update?

Reply:
Currently I look at the OpenPlatformPkg/Drivers/Usb/DwUsb3Dxe/DwUsbDxe.c of HiKey, and found that my patch need to be modified to adjust HiKey. I can do adjusting and verifying and HiKey, but I do not have ArmVExpress and I don't know where is the code of ArmVExpres.

Reply:
EmbeddedPkg/Drivers/Isp1761UsbDxe/